docs: add sitemap setup to Static Site Generation documentation (#505)
analogjs/analog: apps/docs-app/docs/features/server/static-site-generation.md

```

The static pages can be deployed from the `dist/client` directory.

### Sitemap Generation

> Analog also supports automatic sitemap generation. Analog will generate a sitemap in the `dist/analog/public`

Reviewer: ```suggestion
Analog also supports automatic sitemap generation. Analog generates a sitemap in the `dist/analog/public`
```

directory when running a build if a sitemap configuration is provided.

```ts
import { defineConfig } from 'vite';
import analog from '@analogjs/platform';

// https://vitejs.dev/config/
export default defineConfig(({ mode }) => ({
  plugins: [
    analog({
      prerender: {
        routes: async () => ['/', '/blog'],
        sitemap: {

>           host: 'https://analog.org/',

Reviewer: ```suggestion
          host: 'https://analogjs.org/',
```

        },
      },
    }),
  ],
}));
```

> As long as routes are provided, Analog will generate an `.xml` file containing a

Reviewer: ```suggestion
As long as routes are provided, Analog generates a `sitemap.xml` file containing a
```

mapping of the pages' `<loc>` and `<lastmod>` properties.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<urlset...>
    <!--This file was automatically generated by Analog.-->
    <url>

>         <loc>https://analog.org/</loc>

Reviewer: ```suggestion
        <loc>https://analogjs.org/</loc>
```

        <lastmod>2023-07-01</lastmod>
    </url>
    <url>

>         <loc>https://analog.org/blog</loc>

Reviewer: ```suggestion
        <loc>https://analogjs.org/blog</loc>
```
